clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -351,4 +359,13 @@ def ASTFunctionDeclAttr : ASTDecl<"FunctionDecl", "fndecl">;
 def ASTVarDeclAttr : ASTDecl<"VarDecl", "vardecl">;
 def ASTRecordDeclAttr : ASTDecl<"RecordDecl", "recdecl">;
 
+
+def OptnoneAttr : CIR_Unit_Attr<"Optnone", "optnone"> {
+  let storageType = [{ OptnoneAttr }];
+}
+
+def NoInlineAttr : CIR_Attr<"NoInline",  "noInline">  {
+  let storageType = [{ NoInlineAttr }];
+}
```

Comment:
I just got into a place where I also need some `always_inline` information, can you please split off inline handling from this PR? As the title says, this should be about `optnone` :)

I think for now we could have an inline attribute that has two states (enumattr's), `No` and `Always`. The absense of the attribute means a given functions doesn't have any extra inline information.

Reply:
Actually the two attributes are coupled. Without the noline attribute the llvm conversion will fail:

```

Attribute 'optnone' requires 'noinline'!
ptr @_Z2s0ii
fatal error: error in backend: Lowering from LLVMIR dialect to llvm IR failed!
```

Reply:
Sounds good. The noinline attribute is still needed on the llvm IR otherwise the lowering would fail:

```
Attribute 'optnone' requires 'noinline'!
ptr @_Z2s0ii
fatal error: error in backend: Lowering from LLVMIR dialect to llvm IR failed!
```

Reply:
Sounds good, https://github.com/llvm/clangir/pull/134 for inline.